syncopy/tests/test_specest_fooof.py

```diff
@@ -0,0 +1,184 @@
+from syncopy.datatype import SpectralData
+from syncopy.shared.errors import SPYValueError
+from syncopy.tests.test_specest import _make_tf_signal
+
+import matplotlib.pyplot as plt
+
+
+def _plot_powerspec(freqs, powers):
+    """Simple, internal plotting function to plot x versus y."""
+    plt.plot(freqs, powers)
+    plt.xlabel('Frequency (Hz)')
+    plt.ylabel('Power (db)')
+    plt.show()
+
+
+class TestFooofSpy():
+    """
+    Test the frontend (user API) for running FOOOF. FOOOF is a post-processing of an FFT, and
+    to request the post-prcocesing, the user sets the method to "mtmfft", and the output to
+    one of the available FOOOF output types.
+    """
+
+    # Construct input signal
+    nChannels = 2
+    nChan2 = int(nChannels / 2)
+    nTrials = 1
+    seed = 151120
+    fadeIn = None
+    fadeOut = None
+    tfData, modulators, even, odd, fader = _make_tf_signal(nChannels, nTrials, seed,
+                                                           fadeIn=fadeIn, fadeOut=fadeOut, short=True)
+    cfg = get_defaults(freqanalysis)
+    cfg.method = "mtmfft"
+    cfg.taper = "hann"
+    cfg.select = {"trials": 0, "channel": 1}
+    cfg.output = "fooof"
+
+    def test_fooof_output_fooof_fails_with_freq_zero(self, fulltests):
+        """ The fooof package ignores input values of zero frequency, and shortens the output array
+            in that case with a warning. This is not acceptable for us, as the expected output dimension
+            will not off by one. Also it is questionable whether users would want that. We therefore use
+            consider it an error to pass an input frequency axis that contains the zero, and throw an
+            error in the frontend to stop before any expensive computations happen. This test checks for
+            that error.
+        """
+        self.cfg['output'] = "fooof"
+        self.cfg['foilim'] = [0., 250.]    # Include the zero in tfData.
+        self.cfg['out'] = None
+        with pytest.raises(SPYValueError) as err:
+            _ = freqanalysis(self.cfg, self.tfData)  # tfData contains zero.
+        assert "a frequency range that does not include zero" in str(err.value)
+
+    def test_fooof_output_fooof_works_with_freq_zero_in_data_after_setting_foilim(self, fulltests):
+        """
+        This tests the intended operation with output type 'fooof': with an input that does not
+        include zero, ensured by using the 'foilim' argument/setting when calling freqanalysis.
+
+        This returns the full, fooofed spectrum.
+        """
+        self.cfg['output'] = "fooof"
+        self.cfg['foilim'] = [0.5, 250.]    # Exclude the zero in tfData.
+        self.cfg['out'] = None
+        spec_dt = freqanalysis(self.cfg, self.tfData)
+
+        # check frequency axis
+        assert spec_dt.freq.size == 500
+        assert spec_dt.freq[0] == 0.5
+        assert spec_dt.freq[499] == 250.
+
+        # check the log
+        assert "fooof_method = fooof" in spec_dt._log
```

Comment:
Nice! I think we hardly ever test for the log entries directly.. but we should indeed do that explicitly